### PR TITLE
Get evergreen tests to run

### PR DESCRIPTION
**Pull request by andrewsavage1**
Mostly plugs leaks and disables crashing tests.

b/331839823

### CODE REVIEW

File: base/security_unittest.cc [hunk 1]
```diff
@@ -54,8 +54,7 @@ void OverflowTestsSoftExpectTrue(bool overflow_detected) {
     BUILDFLAG(IS_APPLE)
     // Sadly, on Linux, Android, and OSX we don't have a good story yet. Don't
     // fail the test, but report.
-    printf("Platform has overflow: %s\n",
-           !overflow_detected ? "yes." : "no.");
+    printf("Platform has overflow: %s\n", !overflow_detected ? "yes." : "no.");
```
Comment (sherryzy) [inline base/security_unittest.cc:57]: Is this change necessary? If no, maybe we should keep it the same as upstream?

**Review by sherryzy — APPROVED** [review]

**Review by johnxwork — APPROVED** [review]
Great change!

File: base/security_unittest.cc (hunk 1 above)
Comment (andrewsavage1) [reply base/security_unittest.cc:57]: This was accidental, reverted

File: net/BUILD.gn [hunk 2]
```diff
@@ -2858,7 +2858,7 @@ test("net_unittests") {
     "ssl/ssl_config_unittest.cc",
     "ssl/ssl_connection_status_flags_unittest.cc",
     "ssl/ssl_platform_key_util_unittest.cc",
-    "test/embedded_test_server/embedded_test_server_unittest.cc",
+    # "test/embedded_test_server/embedded_test_server_unittest.cc",
```
Comment (sherryzy) [inline net/BUILD.gn:2861]: Do we still need to disable this one? It's passing in my workstation locally.
